unleash/src/main/java/com/silvercar/unleash/repository/FeatureToggleRepository.kt

```diff
@@ -0,0 +1,55 @@
+package com.silvercar.unleash.repository
+
+import com.silvercar.unleash.FeatureToggle
+import com.silvercar.unleash.UnleashException
+import com.silvercar.unleash.event.EventDispatcher
+import com.silvercar.unleash.event.UnleashReady
+import com.silvercar.unleash.repository.FeatureToggleResponse.Status.CHANGED
+import com.silvercar.unleash.util.UnleashConfig
+
+class FeatureToggleRepository(
+  unleashConfig: UnleashConfig,
+  private val toggleFetcher: ToggleFetcher,
+  private val toggleBackupHandler: ToggleBackupHandler
+) : ToggleRepository {
+  private val eventDispatcher: EventDispatcher = EventDispatcher(unleashConfig)
```

Comment:
@deandalm Couldn't this be shortened to `private val eventDispatcher = EventDispatcher(unleashConfig)`?

Reply:
[Kotlin's coding conventions](https://kotlinlang.org/docs/reference/coding-conventions.html#property-formatting) doesn't cover when it's preferred to omit a type or not. 

Are you proposing that if it's very oblivious, we should omit it otherwise keep it (e.g. `private val createdDate: Date = GregorianCalendar(2018, 1, 13).time` we would keep the type)?

Reply:
@deandalm Since Kotlin's coding conventions doesn't specify a preference for an assignment statement, I do propose that we should omit the type declaration in favor of the compiler inferring the type. We should only add it if the type is not obvious.

Reply:
Got it, sounds like a great idea! Do you mind if I open a follow-up PR updating all converted Kotlin files to the new style for properties?

Reply:
That's a good idea. That way we reduce the amount of files to be reviewed in this PR.

Reply:
Is there a way we can tell the linter to look for that type of declaration?

Reply:
Maybe, [Detekt](https://github.com/arturbosch/detekt) has some complex rules that I haven't looked at that deeply.